src/mongocxx/private/database.hpp

```diff
@@ -34,6 +34,19 @@ class database::impl {
         name(std::move(name))
     {}
 
+    impl(const impl& i) : 
+        database_t{libmongoc::database_copy(i.database_t)},
+        client_impl{i.client_impl},
+        name{i.name} 
+    {}
+
+    impl& operator=(const impl& i) {
+        database_t = libmongoc::database_copy(i.database_t);
```

Comment:
I don't think it's true that the destructor is called when you copy assign.  The impl object isn't going out of scope, and its lifetime isn't ending: http://en.cppreference.com/w/cpp/language/destructor

It does seem like we won't ever be using copy assignment.  But if we are defining a copy constructor, it seems like good practice re: the rule of three to also implement copy assignment.  Otherwise we should explicitly delete it with a good comment explaining why we did that.